pandas/tests/frame/common.py

```diff
@@ -139,3 +142,30 @@ def _check_mixed_int(df, dtype=None):
         assert(df.dtypes['C'] == dtypes['C'])
     if dtypes.get('D'):
         assert(df.dtypes['D'] == dtypes['D'])
+
+
+@contextmanager
+def not_raises(expected_exception: Exception,
```

Comment:
I added this to address your call for "the point of tests is to be explicit"  my original code was the minimal example of  a call to `astype` with  the `errors='ignore'` condition that would silence the `ValueError` exception that would otherwise raise can cause a failure.  I'll revert it but I'm left confused by the demand for explicitness and the rejection of this effort to be explicit.

Reply:
Removed in 1c35cc70108a0407d964c26b5cfc07d654dd4ebc